### DISCUSSION TITLE
onDidChangeActiveTextEditor not working for custom text editor

### DISCUSSION THREAD

**Discussion by markwallace-microsoft** [microsoft/vscode-discussions#614] [Extension Development QnA] [answered]
I want to use the `vscode.window.onDidChangeActiveTextEditor` API to get notified when the active text editor changes. This works fine for standard editors but does not get invoked for a custom text editor.

To reproduce the problem I tested using this sample: [custom-editor-sample](https://github.com/microsoft/vscode-extension-samples/tree/main/custom-editor-sample)

I edited [extension.ts](https://github.com/microsoft/vscode-extension-samples/blob/main/custom-editor-sample/src/extension.ts) so the activate methods looks like this:

```
export function activate(context: vscode.ExtensionContext) {
    // Register our custom editor providers
    context.subscriptions.push(CatScratchEditorProvider.register(context));
    context.subscriptions.push(PawDrawEditorProvider.register(context));

    context.subscriptions.push(vscode.window.onDidChangeActiveTextEditor((editor) => {
        console.log("Active Editor Changed: " + editor?.document.fileName);
    }));
}
```
If I open the custom text editors registered by the sample the `onDidChangeActiveTextEditor` callback does not get invoked.

Is this the expected behaviour?

Thanks Mark

**Comment by isidorn** [comment]
@mjbvz might be able to help here

**Comment by jrieken** [comment, marked as answer]
> Is this the expected behaviour?

Yes - the `onDidChangeActiveTextEditor` is about _Text_Editors, not custom editors. You can use `onDidChangeTabGroups` instead

**Reply by trapgar** [reply]
> > Is this the expected behaviour?
> 
> Yes - the `onDidChangeActiveTextEditor` is about _Text_Editors, not custom editors. You can use `onDidChangeTabGroups` instead

Can you clarify if this is intended for custom **text** editors as well? The [custom-editor-sample](https://github.com/microsoft/vscode-extension-samples/tree/main/custom-editor-sample) project has 2 custom editors defined. One of them (`CatScratchEditorProvider`) registers as a `CustomTextEditorProvider`.

```ts
export class CatScratchEditorProvider implements vscode.CustomTextEditorProvider {

	public static register(context: vscode.ExtensionContext): vscode.Disposable {
		const provider = new CatScratchEditorProvider(context);
		const providerRegistration = vscode.window.registerCustomEditorProvider(CatScratchEditorProvider.viewType, provider);

		context.subscriptions.push(vscode.window.onDidChangeActiveTextEditor(editor => {
			if (editor)
				console.log(`Got new CustomTextEditor ${Object.getPrototypeOf(editor).constructor.name}`);
			else
				console.log('You have no open editors or the currently active TextEditor belongs to a different extension.');
		}));

		return providerRegistration;
	}

	// ...
}
```

Just want to confirm that this is the intended design. If so, this should probably be mentioned somewhere in the docs.
